### PR TITLE
Adding version functionality for gateway

### PR DESCRIPTION
**Pull request by krunaljain**
Adding version flag for gateway

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -507,6 +510,7 @@ func (p *gateway) Pipeline() int64 {
 func (p *gateway) configure(loadedConfig *config.GatewayConfig) error {
 	if loadedConfig == nil {
 		return fmt.Errorf("unable to configure gateway with nil config")
+
```
Comment (sstewart-signalfx) [inline main.go]: minor nitpick: remove blank line

File: main_test.go [hunk 2]
```diff
@@ -284,8 +284,14 @@ func Test_Main(t *testing.T) {
...
 		flagParse = flag.Parse
 	}()
+
 	flags.configFileName = "__INVALID_FILENAME__"
 	main()
+
+	// check with the version flag set
+	flags.version = true
+	main()
+
```
Comment (sstewart-signalfx) [inline main_test.go]: nit: remove blank line

File: main_test.go [hunk 3]
```diff
@@ -284,8 +284,13 @@ func Test_Main(t *testing.T) {
 		flags = &gatewayFlags{}
 		flagParse = flag.Parse
 	}()
+
 	flags.configFileName = "__INVALID_FILENAME__"
 	main()
+
+	// check with the version flag set
+	flags.version = true
+	main()
```
Comment (mdubbyap) [inline main_test.go]: So while this certainly satisfy code coverage, it doesn't assert anything and therefore not a very good test. Please assert that the output you expect actually occurred.

Comment (krunaljain) [reply main_test.go]: I added an assert to ensure the version flag toggles from true to false. This happens when we print the version on the console and return

**Pull request merged**